rest/data/interface.go

```diff
@@ -398,20 +398,24 @@ type Connector interface {
 
 	// CompareTasks returns the order that the given tasks would be scheduled, along with the scheduling logic.
 	CompareTasks([]string, bool) ([]string, map[string]map[string]string, error)
+}
 
-	// MockGetGitHubPR Mock Commit queue methods
-	MockGetGitHubPR(context.Context, string, string, int) (*github.PullRequest, error)
-	MockEnqueueItem(string, restModel.APICommitQueueItem, bool) (int, error)
-	MockAddPatchForPr(ctx context.Context, projectRef model.ProjectRef, prNum int, modules []restModel.APIModule, messageOverride string) (string, error)
-	MockFindCommitQueueForProject(string) (*restModel.APICommitQueue, error)
-	MockEnableCommitQueue(*model.ProjectRef) error
-	MockCommitQueueRemoveItem(string, string, string) (*restModel.APICommitQueueItem, error)
-	MockIsItemOnCommitQueue(string, string) (bool, error)
-	MockCommitQueueClearAll() (int, error)
-	MockCreatePatchForMerge(context.Context, string, string) (*restModel.APIPatch, error)
-	MockIsPatchEmpty(string) (bool, error)
-	MockIsAuthorizedToPatchAndMerge(context.Context, *evergreen.Settings, UserRepoInfo) (bool, error)
-	MockGetMessageForPatch(string) (string, error)
-	MockConcludeMerge(string, string) error
-	MockGetAdditionalPatches(patchId string) ([]string, error)
+type MockConnector interface {
+	// Commit queue methods
+	// GetGithubPR takes the owner, repo, and PR number.
+	GetGitHubPR(context.Context, string, string, int) (*github.PullRequest, error)
+	// if bool is true, move the commit queue item to be processed next.
+	EnqueueItem(string, restModel.APICommitQueueItem, bool) (int, error)
+	AddPatchForPr(ctx context.Context, projectRef model.ProjectRef, prNum int, modules []restModel.APIModule, messageOverride string) (string, error)
+	FindCommitQueueForProject(string) (*restModel.APICommitQueue, error)
+	EnableCommitQueue(*model.ProjectRef) error
```

Comment:
A number of these methods don't need to be mocked though, right? EnqueueItem, FindCommitQueueForProject, etc all only touch the DB commit queue

Reply:
How many functions actually require being mocked (would call Github otherwise?) I'm still not convinced we shouldn't just remove those tests. (We should maybe have an in-person sync about this)

Reply:
I've created a new separate mock connector interface that implements these few methods that need to be mocked (the ones talking to GitHub) and implemented them all in one generic mock connector implementation called MockDBConnectorImpl